File: 1/2020.10.08/53.Maximum Subarray/python/solution.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        max_sum = nums[0]
        prev = max_sum
        for i in range(1, len(nums)):
            prev = nums[i] if prev < 0 else prev + nums[i]
            max_sum = max(max_sum, prev)
        return max_sum

File: 1/2020.10.08/53.Maximum Subarray/java/solution.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSum = nums[0];
        int curMax = maxSum;
        for (int i = 1; i < nums.length; i++) {
            curMax = curMax > 0 ? curMax + nums[i] : nums[i];
            maxSum = Math.max(maxSum, curMax);
        }
        return maxSum;
    }
}
